### PR TITLE
🐛 Sort "Backends used by product" list alphabetically

### PR DESCRIPTION
**Pull request by josemigallas**
**What this PR does / why we need it**:

backends have to be sorted by name.

Screenshot 2021-09-07 at 14 42 38

**Which issue(s) this PR fixes** 

Bug reported in: [THREESCALE-6866: [3scale][2.11][LO-prio] Improve list of Backends in Product overview page](https://issues.redhat.com/browse/THREESCALE-6866)

**Verification steps** 

Go to some product overview page and check the list.

### CODE REVIEW

File: app/decorators/service_decorator.rb [hunk 1]
```diff
@@ -68,7 +68,7 @@ def new_application_data
   end
 
   def backends_table_data
-    BackendApiDecorator.decorate_collection(backend_apis)
+    BackendApiDecorator.decorate_collection(backend_apis.sort_by(&:name))
```
Comment (hallelujah) [inline app/decorators/service_decorator.rb:71]: Is backend_apis an `Array` or an `ActiveRecord::Collection` ?

if the latter, then we should use 
```suggestion
    BackendApiDecorator.decorate_collection(backend_apis.desc(name: :asc))
```

Comment (josemigallas) [reply app/decorators/service_decorator.rb:71]: @hallelujah Do you mean
```suggestion
    BackendApiDecorator.decorate_collection(backend_apis.order(name: :asc))
```

Comment (josemigallas) [reply app/decorators/service_decorator.rb:71]: It is a `BackendApi::ActiveRecord_Associations_CollectionProxy` and it doesn't have that sort method.

Comment (hallelujah) [reply app/decorators/service_decorator.rb:71]: I do not get why `order` does not work

```ruby
[2] pry(main)> service.backend_apis.order(name: :asc)                                                                                                                                               
  BackendApi Load (1.4ms)  SELECT `backend_apis`.* FROM `backend_apis` INNER JOIN `backend_api_configs` ON `backend_apis`.`id` = `backend_api_configs`.`backend_api_id` WHERE `backend_api_configs`.`service_id` = 2 ORDER BY `backend_api_configs`.`id` ASC, `backend_apis`.`name` ASC
=> [#<BackendApi:0x000000000cde9958
  id: 2,
  name: "API Backend",
  system_name: "api",
  description: "Backend of API",
  private_endpoint: "https://echo-api.3scale.net:443",
  account_id: 2,
  created_at: Tue, 20 Apr 2021 12:41:24 UTC +00:00,
  updated_at: Tue, 20 Apr 2021 12:41:24 UTC +00:00,
  tenant_id: nil,
  state: "published">]
```

Comment (hallelujah) [inline app/decorators/service_decorator.rb:71]:
```suggestion
    BackendApiDecorator.decorate_collection(backend_apis.reorder(name: :asc))
```

Sorry that was the thing

**Review by hallelujah — CHANGES_REQUESTED** [review]

File: app/decorators/service_decorator.rb (hunk 1 above)
Comment (josemigallas) [reply app/decorators/service_decorator.rb:71]: Done

**Review by thomasmaas — APPROVED** [review]
You're using `reorder` because our default ordering is different (I guess). Changing the default doesn't make sense?

**Comment by josemigallas** [comment]
> You're using `reorder` because our default ordering is different (I guess). Changing the default doesn't make sense?

I'd rather set a specific order at the view layer, where we need rather than changing the default order and unexpectedly affect other views.

**Review by hallelujah — APPROVED** [review]

**Pull request merged**